src/test/java/com/connexta/search/common/SearchITests.java

```diff
@@ -358,27 +383,57 @@ public void testQueryWhenSolrIsEmpty() throws Exception {
         (List<URI>) restTemplate.getForObject(queryUriBuilder.build(), List.class), is(empty()));
   }
 
-  private static HttpEntity createIndexRequest(final String fileString) throws IOException {
-    // TODO replace with request class from api dependency
-    final InputStream metadataInputStream = IOUtils.toInputStream(fileString, "UTF-8");
-    final MultiValueMap<String, Object> requestBody = new LinkedMultiValueMap<>();
-    requestBody.add(
-        "file",
-        new InputStreamResource(metadataInputStream) {
+  @Test
+  public void testIndexingAndQueryingAllAttributes() throws IOException {
+    Map<String, String> doc = getSampleDatatHavingAllAttributes();
 
-          @Override
-          public long contentLength() throws IOException {
-            return metadataInputStream.available();
-          }
+    // Assert valid preconditions
+    assertThat(
+        "Sample data must include all query attributes",
+        unmodifiableSet(doc.keySet()),
+        equalTo(QUERY_TERMS));
+
+    // Index the document
+    indexManager.index(
+        doc.get(ID_ATTRIBUTE_NAME),
+        doc.get(MEDIA_TYPE_ATTRIBUTE_NAME),
+        IOUtils.toInputStream("{ \"ext.extracted.text\" : \"Winterfell\" }", "UTF-8"));
+
+    // Query for the document
+    String q = getQueryForAllAttributes();
+    List<URI> results = queryManager.find(q);
+    MatcherAssert.assertThat("Expected exactly one result", results, hasSize(1));
+  }
 
-          @Override
-          public String getFilename() {
-            // The extension of this filename is used to get the ContentType of the file.
-            return "ignored.json";
-          }
-        });
-    final HttpHeaders httpHeaders = new HttpHeaders();
-    httpHeaders.set("Accept-Version", "0.2.0");
-    return new HttpEntity<>(requestBody, httpHeaders);
+  @NotNull
+  private Map<String, String> getSampleDatatHavingAllAttributes() {
```

Comment:
✏️`static`
✏️Does this need to be a method? Would it be simpler to use of a `private static final` field because it's not dynamic?

(same comments for `getQueryForAllAttributes`)

Reply:
Good suggestion. They look better as vars.